frontend/ui/lib/widgets/place_block_widget.dart

```diff
@@ -1,75 +1,232 @@
 import 'package:flutter/material.dart';
+import 'package:tripmeout/model/place.dart';
+import 'package:tripmeout/model/place_visit.dart';
+import 'package:tripmeout/services/place_visit_service.dart';
+import 'package:url_launcher/url_launcher.dart';
 
 class PlaceBlockWidget extends StatefulWidget {
-  final String placeName;
+  final PlaceVisit placeVisit;
+  final PlaceVisitService placeVisitService;
+  final PlaceWrapper details;
 
-  PlaceBlockWidget(this.placeName);
+  PlaceBlockWidget(this.placeVisit, this.placeVisitService, this.details);
 
   @override
-  State createState() => _PlaceBlockWidgetState(placeName);
+  State createState() =>
+      _PlaceBlockWidgetState(placeVisit, placeVisitService, details);
 }
 
 class _PlaceBlockWidgetState extends State<PlaceBlockWidget> {
-  final String placeName;
-  _PlaceBlockWidgetState(this.placeName);
-
-  List<Widget> pictures = new List<Widget>();
-  List<Color> colors = [
-    Colors.red,
-    Colors.blue,
-    Colors.green,
-    Colors.yellow,
-    Colors.orange,
-  ];
+  final PlaceVisit placeVisit;
+  final PlaceVisitService placeVisitService;
+  final PlaceWrapper details;
+  _PlaceBlockWidgetState(this.placeVisit, this.placeVisitService, this.details);
 
   bool _selected = false;
   Icon _icon = Icon(Icons.favorite_border);
   Color _color = Colors.black;
 
   @override
   Widget build(BuildContext context) {
```

Comment:
This method is wayyyyyy too big. But I don't want to style pick and the final product looks good.